### PR TITLE
Tl dsdegp 1740 regulatory designation

### PR DESCRIPTION
**Pull request by tlangs**
### Description

Added regulatory designation to WgsCrams and Gvcfs. In both, the regulatory designation is automatically set to RESEARCH_ONLY in the respective service classes.

_Please explain the changes you made here._
_Explain the **motivation** for making this change. What existing problem does the pull request solve?_
_Mention any JIRA issues fixed, addressed or otherwise related to this pull request, including issue numbers or hard links for issues in other repos._

----

### Checklist (never delete this)

Never delete this, it is our record that procedure was followed. If you find that for whatever reason one of the checklist points doesn't apply to your PR, you can leave it unchecked but please add an explanation below.

- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: Ensure that you have added or updated tests
- [ ] **Submitter**: If you're adding/switching libraries or otherwise changing the design, update the [design documentation](https://broadinstitute.atlassian.net/wiki/pages/viewpage.action?pageId=114531509).
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * Add notes on what you've tested
* Review cycle:
  * Team may comment on PR at will
  * **Reviewer assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to reviewer** for further feedback
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Move the JIRA issue to QA once this checklist is completed

### CODE REVIEW

File: clio-dataaccess-model/src/main/scala/org/broadinstitute/clio/server/dataaccess/elasticsearch/DocumentGvcf.scala [hunk 1]
```diff
@@ -21,5 +26,7 @@ case class DocumentGvcf(upsertId: UpsertId,
                         gvcfSummaryMetricsPath: Option[URI] = None,
                         gvcfDetailMetricsPath: Option[URI] = None,
                         pipelineVersion: Option[Symbol] = None,
+                        regulatoryDesignation: Option[RegulatoryDesignation] =
+                          None,
```
Comment (danxmoran) [inline clio-dataaccess-model/src/main/scala/org/broadinstitute/clio/server/dataaccess/elasticsearch/DocumentGvcf.scala:30]: Curse ye scalafmt

File: clio-integration-test/src/it/scala/org/broadinstitute/clio/integrationtest/tests/GvcfTests.scala [hunk 2]
```diff
@@ -78,7 +83,8 @@ trait GvcfTests { self: BaseIntegrationSpec =>
       sampleAlias = s"someAlias$randomId",
       version = 2,
       documentStatus = Some(DocumentStatus.Normal),
-      gvcfPath = Some(URI.create("gs://path/gvcf.gvcf"))
+      gvcfPath = Some(URI.create("gs://path/gvcf.gvcf")),
+      regulatoryDesignation = Some(RegulatoryDesignation.ResearchOnly)
```
Comment (danxmoran) [inline clio-integration-test/src/it/scala/org/broadinstitute/clio/integrationtest/tests/GvcfTests.scala:87]: Could this be something other than `ResearchOnly`, to show that the server doesn't overwrite the value if it's given?

File: clio-integration-test/src/it/scala/org/broadinstitute/clio/integrationtest/tests/WgsCramTests.scala [hunk 3]
```diff
@@ -74,6 +79,7 @@ trait WgsCramTests { self: BaseIntegrationSpec =>
       sampleAlias = s"someAlias$randomId",
       version = 2,
       documentStatus = Some(DocumentStatus.Normal),
+      regulatoryDesignation = Some(RegulatoryDesignation.ResearchOnly),
```
Comment (danxmoran) [inline clio-integration-test/src/it/scala/org/broadinstitute/clio/integrationtest/tests/WgsCramTests.scala:82]: Same here.

File: clio-server/src/main/scala/org/broadinstitute/clio/server/service/GvcfService.scala [hunk 4]
```diff
@@ -31,7 +35,8 @@ class GvcfService(
   ): Future[UpsertId] = {
     val updatedTransferMetadata = transferMetadata.copy(
       documentStatus =
-        transferMetadata.documentStatus.orElse(Some(DocumentStatus.Normal))
+        transferMetadata.documentStatus.orElse(Some(DocumentStatus.Normal)),
+      regulatoryDesignation = Some(RegulatoryDesignation.ResearchOnly)
```
Comment (danxmoran) [inline clio-server/src/main/scala/org/broadinstitute/clio/server/service/GvcfService.scala]: This should be `transferMetadata.orElse(...)`

File: clio-server/src/main/scala/org/broadinstitute/clio/server/service/WgsCramService.scala [hunk 5]
```diff
@@ -31,7 +35,8 @@ class WgsCramService(
   ): Future[UpsertId] = {
     val updatedTransferMetadata = transferMetadata.copy(
       documentStatus =
-        transferMetadata.documentStatus.orElse(Some(DocumentStatus.Normal))
+        transferMetadata.documentStatus.orElse(Some(DocumentStatus.Normal)),
+      regulatoryDesignation = Some(RegulatoryDesignation.ResearchOnly)
```
Comment (danxmoran) [inline clio-server/src/main/scala/org/broadinstitute/clio/server/service/WgsCramService.scala]: Same here.

**Review by danxmoran — CHANGES_REQUESTED** [review]

File: clio-integration-test/src/it/scala/org/broadinstitute/clio/integrationtest/tests/WgsCramTests.scala [hunk 6]
```diff
@@ -856,6 +863,54 @@ trait WgsCramTests { self: BaseIntegrationSpec =>
     }
   }
 
+  it should "automatically set regulatory designation to RESEARCH_ONLY for crams" in {
```
Comment (pshapiro4broad) [inline clio-integration-test/src/it/scala/org/broadinstitute/clio/integrationtest/tests/WgsCramTests.scala]: Should probably spell "research only" the same way it's done in the code, e.g. `ResearchOnly`.

File: clio-integration-test/src/it/scala/org/broadinstitute/clio/integrationtest/tests/WgsCramTests.scala [hunk 7]
```diff
@@ -856,6 +863,54 @@ trait WgsCramTests { self: BaseIntegrationSpec =>
...
+      rootTestStorageDir.resolve(s"cram/$project/$sample/v$version/")
+    val cramSource = rootSource.resolve(cramName)
+    val craiSource = rootSource.resolve(craiName)
+
+    val key = TransferWgsCramV1Key(Location.GCP, project, sample, version)
+    val metadata = TransferWgsCramV1Metadata(
+      cramPath = Some(cramSource.toUri),
+      craiPath = Some(craiSource.toUri),
+      cramMd5 = Some(Symbol(md5Contents)),
+      regulatoryDesignation = None
```
Comment (pshapiro4broad) [inline clio-integration-test/src/it/scala/org/broadinstitute/clio/integrationtest/tests/WgsCramTests.scala:886]: can't you omit this line?

**Review by pshapiro4broad — APPROVED** [review]
This looks OK, but it seems like it would've been a lot easier to test this if the code that modifies the metadata was done in a separate method, eg. `updateTransferMetadata()`. That way you could test just this behavior without having to do an upsert, round trip, etc.

File: clio-integration-test/src/it/scala/org/broadinstitute/clio/integrationtest/tests/GvcfTests.scala (hunk 2 above)
Comment (tlangs) [reply clio-integration-test/src/it/scala/org/broadinstitute/clio/integrationtest/tests/GvcfTests.scala:87]: This is the expected value we want to get back from the server

Comment (tlangs) [reply clio-integration-test/src/it/scala/org/broadinstitute/clio/integrationtest/tests/GvcfTests.scala]: This is the expected GVCF that the GVCF returned from the server is being tested against.

File: clio-integration-test/src/it/scala/org/broadinstitute/clio/integrationtest/tests/WgsCramTests.scala (hunk 3 above)
Comment (tlangs) [reply clio-integration-test/src/it/scala/org/broadinstitute/clio/integrationtest/tests/WgsCramTests.scala:82]: Same here.

File: clio-integration-test/src/it/scala/org/broadinstitute/clio/integrationtest/tests/WgsCramTests.scala (hunk 7 above)
Comment (tlangs) [reply clio-integration-test/src/it/scala/org/broadinstitute/clio/integrationtest/tests/WgsCramTests.scala:886]: I'm explicitly setting it to None to make it clear in the test that the value should be changed by the server.

File: clio-server/src/main/scala/org/broadinstitute/clio/server/service/WgsCramService.scala (hunk 5 above)
Comment (tlangs) [reply clio-server/src/main/scala/org/broadinstitute/clio/server/service/WgsCramService.scala]: You're right, that will mean we don't have to make a change in the future.

File: clio-integration-test/src/it/scala/org/broadinstitute/clio/integrationtest/tests/GvcfTests.scala (hunk 2 above)
Comment (danxmoran) [reply clio-integration-test/src/it/scala/org/broadinstitute/clio/integrationtest/tests/GvcfTests.scala:87]: Ah, missed that this was in the `expected`. Could we still change this to something else, and then pass the same something else in with the metadata? This sort of test would've caught the other problem I commented on.

Comment (tlangs) [reply clio-integration-test/src/it/scala/org/broadinstitute/clio/integrationtest/tests/GvcfTests.scala:87]: like this? https://github.com/broadinstitute/clio/pull/164/files/802a5c9dd95f5a1f5973736c1ca7d7f2f04de913#diff-a369a91bf3d67fb7ffd26a123a61aa14R795
(EDIT: Dammit I did it in WgsCram but not in Gvcfs)

Comment (danxmoran) [reply clio-integration-test/src/it/scala/org/broadinstitute/clio/integrationtest/tests/GvcfTests.scala:87]: No, as in a test where you set it to something other than RESEARCH_ONLY and it isn't changed by the server.

Comment (tlangs) [reply clio-integration-test/src/it/scala/org/broadinstitute/clio/integrationtest/tests/GvcfTests.scala:87]: ah ok.

File: clio-integration-test/src/it/scala/org/broadinstitute/clio/integrationtest/tests/GvcfTests.scala [hunk 8]
```diff
@@ -783,6 +793,79 @@ trait GvcfTests { self: BaseIntegrationSpec =>
...
+
+    val cloudPath = rootTestStorageDir.resolve(
+      s"gvcf/$project/$sample/v$version/$randomId.gvcf"
+    )
+
+    val key = TransferGvcfV1Key(Location.GCP, project, sample, version)
+    val metadata =
+      TransferGvcfV1Metadata(
+        gvcfPath = Some(cloudPath.toUri),
+        regulatoryDesignation = Some(ClinicalDiagnostics)
```
Comment (danxmoran) [inline clio-integration-test/src/it/scala/org/broadinstitute/clio/integrationtest/tests/GvcfTests.scala]: Nitpick: all the other usages have `RegulatoryDesignation.` in front of the enum value.

**Review by danxmoran — APPROVED** [review]

**Comment by tlangs** [comment]
retest this please

**Pull request merged**